Library/Homebrew/utils.rb

```diff
@@ -175,7 +175,17 @@ def quiet_system cmd, *args
 end
 
 def curl *args
-  curl = Pathname.new '/usr/bin/curl'
+  brewed_curl = begin
+    Formula["curl"]
+  rescue FormulaUnavailableError
+    nil
+  end
+  use_curl = MacOS.version <= "10.6" && !brewed_curl.nil? && brewed_curl.installed?
+  if use_curl
+    curl = brewed_curl.opt_bin/"curl"
+  else
+    curl = Pathname.new '/usr/bin/curl'
+  end
```

Comment:
All of the code related to homebrew's curl should be inside the `MacOS.version <= "10.6"` conditional, as there is no reason for it to ever execute otherwise.

When would we expect `Formula["curl"]` to raise an exception? The formula will always be present.